src/main/kotlin/no/nav/helsearbeidsgiver/bro/sykepenger/TilgjengeliggjoerForespoerslerForVedtaksperiodeIdListeRiver.kt

```diff
@@ -0,0 +1,99 @@
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helse.rapids_rivers.River
+import no.nav.helsearbeidsgiver.bro.sykepenger.db.ForespoerselDao
+import no.nav.helsearbeidsgiver.bro.sykepenger.domene.ForespoerselSimba
+import no.nav.helsearbeidsgiver.bro.sykepenger.domene.HentForespoerslerForVedtaksperiodeIdListeSvar
+import no.nav.helsearbeidsgiver.bro.sykepenger.kafkatopic.pri.Pri
+import no.nav.helsearbeidsgiver.bro.sykepenger.kafkatopic.pri.PriProducer
+import no.nav.helsearbeidsgiver.bro.sykepenger.utils.Loggernaut
+import no.nav.helsearbeidsgiver.bro.sykepenger.utils.demandValues
+import no.nav.helsearbeidsgiver.bro.sykepenger.utils.les
+import no.nav.helsearbeidsgiver.bro.sykepenger.utils.rejectKeys
+import no.nav.helsearbeidsgiver.bro.sykepenger.utils.requireKeys
+import no.nav.helsearbeidsgiver.utils.json.fromJsonMapFiltered
+import no.nav.helsearbeidsgiver.utils.json.parseJson
+import no.nav.helsearbeidsgiver.utils.json.serializer.UuidSerializer
+import no.nav.helsearbeidsgiver.utils.json.serializer.list
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import no.nav.helsearbeidsgiver.utils.json.toPretty
+import java.util.UUID
+
+// Tilgjengeliggjør eksponerte forespørsler for en liste med vedtaksperiode-IDer
+class TilgjengeliggjoerForespoerslerForVedtaksperiodeIdListeRiver(
+    rapid: RapidsConnection,
+    private val forespoerselDao: ForespoerselDao,
+    private val priProducer: PriProducer,
+) : River.PacketListener {
+    private val loggernaut = Loggernaut(this)
+
+    init {
+        River(rapid)
+            .apply {
+                validate { msg ->
+                    msg.demandValues(
+                        Pri.Key.BEHOV to HentForespoerslerForVedtaksperiodeIdListeSvar.behovType.name,
+                    )
+                    msg.requireKeys(Pri.Key.BOOMERANG, Pri.Key.VEDTAKSPERIODE_ID_LISTE)
+                    msg.rejectKeys(Pri.Key.LØSNING)
+                }
+            }.register(this)
+    }
+
+    override fun onPacket(
+        packet: JsonMessage,
+        context: MessageContext,
+    ) {
+        val json = packet.toJson().parseJson()
+
+        val vedtaksperiodeIdListe =
+            Pri.Key.VEDTAKSPERIODE_ID_LISTE.les(
+                UuidSerializer.list(),
+                json.fromJsonMapFiltered(Pri.Key.serializer()),
+            )
+
+        runCatching {
+            json.sendSvar(vedtaksperiodeIdListe)
+        }.onFailure(loggernaut::ukjentFeil)
+    }
+
+    private fun JsonElement.sendSvar(vedtaksperiodeIdListe: List<UUID>) {
+        val melding = fromJsonMapFiltered(Pri.Key.serializer())
+
+        "Mottok melding på pri-topic".also {
+            loggernaut.aapen.info("$it av type '${Pri.Key.BEHOV.les(String.serializer(), melding)}'.")
+            loggernaut.sikker.info("$it med innhold:\n${toPretty()}")
+        }
+
+        val forespoersler = vedtaksperiodeIdListe.mapNotNull { forespoerselDao.hentForespoerselEksponertTilSimba(it) }
```

Comment:
Ikke særlig effektivt. Tror det hadde gjort seg bedre med en dedikert spørring for flere vedtaksperiode-ID-er.

Reply:
Gjort om til en spørring som takler en liste med vedtaksperiode-IDer nå, og tatt i bruk denne spørringen også dersom man bare er ut etter én forespørsel.